lib/common/validation.ts

```diff
@@ -65,7 +90,68 @@ export function getStrength(password: string): PasswordStrength {
   };
 }
 
+// password validation
+enum PasswordCriteria {
+  ConfirmPassword = 'confirmPassword',
+  Length = 'length',
+  Uppercase = 'uppercase',
+  Lowercase = 'lowercase',
+  Number = 'number',
+  Special = 'special',
+}
+
+export const strengthCriteria = [
+  PasswordCriteria.Length,
+  PasswordCriteria.Uppercase,
+  PasswordCriteria.Lowercase,
+  PasswordCriteria.Number,
+  PasswordCriteria.Special,
+];
+
+const MIN_PASSWORD_LENGTH = 12;
+
+export function doPasswordMatch(
```

Comment:
I find this function super weird and unintuitive. Its name is wrong, it takes an array of enum values used as flags, it probably shouldn't also check if the confirmPassword and the password are the same.

Is there some specific reason it's like this? Something like this would seem more practical:

```typescript
interface PasswordCriteria {
  testLength?: boolean;
  testUppercase?: boolean;
  testDigit?: boolean;
  ...
}

const PasswordCriteriaAll {
  testLength: true,
  testUppercase: true,
  testDigit: true,
  ...
};

function passwordCriteriaMatch(password: string, criteria: PasswordCriteria): boolean {
  let res = true;
  if (res && criteria.testLength) {
    res &&= password.length > MIN_LENGTH;
  }
  if (res && criteria.testUppercase) {
    res &&= hasUppercase(password);
  }
  ...
  return res;
}

passwordCriteriaMatch('P@ssw0rd', { testLength: true, testDigit: true });
passwordCriteriaMatch('P@ssw0rd', PasswordCriteriaAll);
```